packages/payment-backend/src/service.ts

```diff
@@ -262,12 +280,15 @@ export const getPaymentMethodList = async (parameters: {
 
   const globallyFilteredPaymentGateways = gateways.split(',')
 
-  filteredPaymentFilters = filteredPaymentFilters.filter((method) => {
-    return (
-      !globallyFilteredPaymentMethods.includes(method.code.toLowerCase()) ||
-      !globallyFilteredPaymentGateways.includes(method.gateway.toLowerCase())
-    )
-  })
+  filteredPaymentFilters = filterPaymentMethodByGateway(
```

Comment:
At the moment these functions dont return a filtered array, but rather you will have the initial "filteredPaymentFilters" array, because the filter methods dont actually return the filtered arrays.

Reply:
Never mind this, I did not notice its lambda definition. Good job here!